components/pkg-export-tar/src/rootfs.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs;
+use std::path::Path;
+
+use hcore::os::filesystem;
+
+use error::Result;
+
+/// Creates a root file system under the given path.
+///
+/// # Errors
+///
+/// * If files and/or directories cannot be created
+/// * If permissions for files and/or directories cannot be set
+pub fn create<T>(root: T) -> Result<()>
+where
+    T: AsRef<Path>,
+{
+    let root = root.as_ref();
+    fs::create_dir_all(root.join("bin"))?;
+    fs::create_dir_all(root.join("etc"))?;
+
+    fs::create_dir_all(root.join("root"))?;
+    filesystem::chmod(root.join("root").to_string_lossy().as_ref(), 0o0750)?;
+    fs::create_dir_all(root.join("tmp"))?;
+    filesystem::chmod(root.join("tmp").to_string_lossy().as_ref(), 0o1777)?;
+    fs::create_dir_all(root.join("var/tmp"))?;
+    filesystem::chmod(root.join("var/tmp").to_string_lossy().as_ref(), 0o1777)?;
+
+    Ok(())
```

Comment:
I don't think we need `/bin`, `/etc`, `/tmp`, or `/var` in a tar export, assuming that a user will be expanding the tarball onto an existing Linux system, rather than having to "carry one around with us" in a container scenario.

(In a future with a finalized `BuildRoot` abstraction, the Docker exporter would probably implement some kind of `BuildRoot` trait that specializes this behavior, whereas the tar exporter would probably just use a default no-op implementation.)

Reply:
Fixed!